### PR TITLE
Time-series Profit-and-Loss

### PR DESCRIPTION
**Pull request by pgebheim**

### CODE REVIEW

File: src/server/getters/get-profit-loss.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
+import * as Knex from "knex";
+import * as _ from "lodash";
+import { mapValues } from "async";
```
Comment (epheph) [inline src/server/getters/get-profit-loss.ts]: mapValues, AsyncCallback, CalculatedProfitLoss unused

File: src/server/getters/get-profit-loss.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+function sub(n1: string, n2: string) {
+  return new BigNumber(n1, 10).minus(new BigNumber(n2));
+}
+
+function times(n1: string, n2: string) {
+  return new BigNumber(n1, 10).times(new BigNumber(n2));
+}
+
```
Comment (epheph) [inline src/server/getters/get-profit-loss.ts]: times and div unused

File: src/server/getters/get-profit-loss.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    // If the insertPoint is zero, then we don't have any "lastPrice" value -- e.g. there are
+    // no trades in that point which will result in a `null` PL.
+    const insertPoint: number = _.sortedIndexBy(outcomeTrades, { timestamp: bucket.timestamp }, (trade) => trade.timestamp);
+    if (insertPoint > 0) {
+      return Object.assign({}, bucket, { lastPrice: outcomeTrades[insertPoint - 1].price!.toFixed() });
+    }
+
+    return bucket;
+  });
+  return bucketsWithLastPrice;
```
Comment (epheph) [inline src/server/getters/get-profit-loss.ts]: just return it? Tiny nit

**Comment by epheph** [comment]
This looks great. There's a couple cleanup things i mentioned, and a couple `console.log()`s in there, but otherwise this is :godmode:

**Comment by pgebheim** [comment]
Ok! I'll clean up the console logs (duh)!

**Pull request merged**